File: 477.Total Hamming Distance.cpp
class Solution {
public:
    int totalHammingDistance(vector<int>& nums) {
        int total = 0;
        for(int shift=0;shift<31;shift++){
            int mask = 1<<shift;
            int bitSet = 0;
            for(auto num : nums){
                bitSet+=(num&mask)?1:0;
            }
            total+=bitSet*(nums.size()-bitSet);
        }
        return total;
    }
};

File: 477.Total Hamming Distance.py
class Solution:
    def totalHammingDistance(self, nums: List[int]) -> int:
        res = 0
        for shift in range(30):
            cnt = 0
            for num in nums:
                cnt += (num >> shift) & 1
            res += cnt * (len(nums) - cnt)
        return res
